src/utils/assert.ts

```diff
@@ -61,6 +61,10 @@ export const assert = {
     isBoolean(variableName: string, value: boolean): void {
         this.assert(_.isBoolean(value), this.typeAssertionMessage(variableName, 'boolean', value));
     },
+    isWeb3Provider(variableName: string, value: Web3.Provider): void {
+        const isWeb3Provider = _.isFunction((value as any).send) || _.isFunction((value as any).sendAsync);
```

Comment:
Does the `Web3.Provider` type not have the `send` or `sendAsync` methods defined on it? Why is `any` necessary here?

Reply:
Currently it's an empty interface. I can update web3 types